cmd/gardener-seed-admission-controller/app/gardener_seed_admission_controller.go

```diff
@@ -0,0 +1,241 @@
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+	admissionregistrationv1beta1.AddToScheme(scheme)
+	deserializer := serializer.NewCodecFactory(scheme).UniversalDeserializer()
+
+	mux := http.NewServeMux()
+
+	k8sClient, err := kubernetes.NewClientFromFile("", kubeconfigPath, kubernetes.WithClientOptions(client.Options{
+		Scheme: kubernetes.SeedScheme,
+	}))
+	if err != nil {
+		logger.Errorf("unable to create kubernetes client: %+v", err)
+		panic(err)
+	}
+
+	seedAdmissionController := &GardenerSeedAdmissionController{
+		deserializer,
+		k8sClient.Client(),
+	}
+	mux.HandleFunc("/webhooks/validate-extension-crd-deletion", seedAdmissionController.validateExtensionCRDDeletion)
+
+	srv := &http.Server{
+		Addr:    fmt.Sprintf("%s:%d", bindAddress, port),
+		Handler: mux,
+	}
+
+	go func() {
+		logger.Infof("Starting Gardener seed admission controller...")
+		if err := srv.ListenAndServeTLS(certPath, keyPath); err != http.ErrServerClosed {
+			logger.Errorf("Could not start HTTPS server: %v", err)
+			panic(err)
+		}
+	}()
+
+	<-ctx.Done()
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	if err := srv.Shutdown(timeoutCtx); err != nil {
+		logger.Errorf("Error when shutting down HTTPS server: %v", err)
+	}
+	logger.Info("HTTPS servers stopped.")
+}
+
+func respond(w http.ResponseWriter, response *admissionv1beta1.AdmissionResponse) {
+	responseObj := admissionv1beta1.AdmissionReview{}
+	if response != nil {
+		responseObj.Response = response
+	}
+
+	jsonResponse, err := json.Marshal(responseObj)
+	if err != nil {
+		logger.Error(err)
+	}
+	if _, err := w.Write(jsonResponse); err != nil {
+		logger.Error(err)
+	}
+}
+
+// GardenerSeedAdmissionController represents all the parameters required to start the
+// Gardener seed admission controller.
+type GardenerSeedAdmissionController struct {
+	codecs runtime.Decoder
+	client client.Client
+}
+
+func (g *GardenerSeedAdmissionController) handleAdmissionReview(w http.ResponseWriter, r *http.Request) (admissionv1beta1.AdmissionReview, error) {
+	var (
+		body           []byte
+		receivedReview = admissionv1beta1.AdmissionReview{}
+	)
+
+	// Read HTTP request body into variable.
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			body = data
+		}
+	}
+
+	// Verify that the correct content-type header has been sent.
+	if contentType := r.Header.Get("Content-Type"); contentType != "application/json" {
+		err := fmt.Errorf("contentType=%s, expect %s", contentType, "application/json")
+		logger.Errorf(err.Error())
+		respond(w, &admissionv1beta1.AdmissionResponse{Result: &metav1.Status{Message: err.Error()}})
+		return receivedReview, err
+	}
+
+	// Deserialize HTTP request body into admissionv1beta1.AdmissionReview object.
+	if _, _, err := g.codecs.Decode(body, nil, &receivedReview); err != nil {
+		logger.Errorf(err.Error())
```

Comment:
Can you also add a error message here?
```suggestion
		err = fmt.Errorf("invalid request body (error decoding body): %+v", err)
		logger.Errorf(err.Error())
```